File: 1047. Remove All Adjacent Duplicates In String/solution.java
class Solution {
    public String removeDuplicates(String s) {
        StringBuilder builder = new StringBuilder();
        for(int i=0;i<s.length();++i){
            if(builder.length() !=0 && builder.charAt(builder.length()-1)==s.charAt(i)){
                builder.deleteCharAt(builder.length()-1);
                continue;
            }
            builder.append(s.charAt(i));
        }
        return builder.toString();
    }
}

File: 1047. Remove All Adjacent Duplicates In String/solution.py
class Solution(object):
    def removeDuplicates(self, s):
        index = 0
        for element in s:
                    if(index == 0):
                        index+=1
                        continue
                    elif(s[index-1] == element):
                        s = s[:index-1] + s[index+1:]
                        index-=1
                    else:
                        index+=1
        return s